qiime2/sdk/plugin_manager.py

```diff
@@ -38,29 +38,57 @@ def iter_entry_points(cls):
 
     # This class is a singleton as it is slow to create, represents the
     # state of a qiime2 installation, and is needed *everywhere*
-    def __new__(cls):
+    def __new__(cls, install_plugins=True):
         if cls.__instance is None:
             self = super().__new__(cls)
-            self._init()
+            self._init(install_plugins=install_plugins)
             cls.__instance = self
         return cls.__instance
 
-    def _init(self):
+    def _init(self, install_plugins):
         self.plugins = {}
+        self._plugin_by_id = {}
         self.semantic_types = {}
         self.transformers = collections.defaultdict(dict)
         self.formats = {}
         self.views = {}
         self.type_formats = []
 
-        # These are all dependent loops, each requires the loop above it to
-        # be completed.
-        for entry_point in self.iter_entry_points():
-            plugin = entry_point.load()
-            self.plugins[plugin.name] = plugin
-
-        for plugin in self.plugins.values():
-            self._integrate_plugin(plugin)
+        if install_plugins:
+            # These are all dependent loops, each requires the loop above it to
+            # be completed.
+            for entry_point in self.iter_entry_points():
+                project_name = entry_point.dist.project_name
+                package = entry_point.module_name.split('.')[0]
+                plugin = entry_point.load()
+
+                self.install_plugin(plugin, package, project_name)
+
+    def install_plugin(self, plugin, package=None, project_name=None):
```

Comment:
I wonder if there's another word we can use. Since installing things is usually such a challenge they are probably not going to think of this method in the context of adding it to an instance.

What about `add_plugin` and `add_plugins=True`?

Reply:
Good point. I like `add` vs `install`.